src/store/modules/emails.js

```diff
@@ -1,27 +1,95 @@
-const state = [
-  {
-    senderId: 1,
-    senderName: "Jerry",
-    senderIcon: "logo.png",
-    emails: [
+const state = {
+  selectedSenderId: 1,
+};
+
+// getters
+const getters = {
+  getEmails() {
+    //this would use the email service
+    let mail = [
       {
-        emailId: 1,
-        receivedTime: Date.now(),
-        messageTitle: "New Inbox Message!",
-        messageBody: "Lorem Ipsum",
+        senderId: 1,
+        senderName: "Jerry",
+        senderIcon: "logo.png",
+        emails: [
+          {
+            emailId: 1,
+            receivedTime: new Date("May 14, 2021 10:04:00"),
+            messageTitle:
+              "Your Employment Insurance Claim - You've been subscribed to Job Alerts",
+            messageBody: `You are receiving this email because you recently completed an application for Employment Insurance. When you filed your claim, you agreed to be actively looking for work. To help you with your job search, we’ve subscribed you to Job Alerts.
+    
+    Thank you,
+    
+    Job Bank`,
+          },
+          {
+            emailId: 2,
+            receivedTime: new Date(),
+            messageTitle: "New Inbox Message!",
+            messageBody: "Lorem Ipsum",
+          },
+        ],
       },
-    ],
+    ];
+
+    return mail;
   },
-];
+  getSelectedSenderObject(state, getters) {
+    return getters.getEmails.find(
+      (sender) => sender.senderId === state.selectedSenderId
+    );
+  },
+  //returns all the emails, with most recent ones first
+  getEmailsOrderByDateAsc(state, getters) {
+    //sorts a copy of the list of mail (getters should not mutate state)
+    const mail = getters.getSelectedSenderObject.emails;
+    const sortedMail = mail.sort((a, b) => {
+      if (a.receivedTime > b.receivedTime) {
+        return -1;
+      }
+      if (a.receivedTime < b.receivedTime) {
+        return 1;
+      }
+      return 0;
+    });
 
-// getters
-const getters = {};
+    return sortedMail;
+  },
+};
 
 // actions
-const actions = {};
+const actions = {
+  //these functions would probably be used by the email service
+  sendEmail({ commit }, payload) {
```

Comment:
I do not know if we will ever be able to "send emails" or "add emails". This logic might be useless for a while. (correct me if i am wrong please.)

Reply:
That is true, I originally thought that there might be a mechanism on the client-side to add emails but obviously that'd be handled by the email service. Given that it's a very simple implementation, I'll remove it and if we ever need to, it won't be a big issue to re-add it.